impl/sql/mutationstorage/queue_test.go

```diff
@@ -40,6 +40,52 @@ func newForTest(ctx context.Context, t testing.TB, logIDs ...int64) *Mutations {
 	return m
 }
 
+func TestSetWritable(t *testing.T) {
+	ctx := context.Background()
+
+	for _, tc := range []struct {
+		desc       string
+		logIDs     []int64
+		set        map[int64]bool
+		wantLogIDs []int64
+		wantCode   codes.Code
+	}{
+		{desc: "one row", logIDs: []int64{10}, wantLogIDs: []int64{10}},
+		{desc: "one row disabled", logIDs: []int64{10}, set: map[int64]bool{10: false}, wantCode: codes.NotFound},
+		{desc: "multi", logIDs: []int64{1, 2, 3}, set: map[int64]bool{1: true, 2: false}, wantLogIDs: []int64{1, 3}},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			m := newForTest(ctx, t, tc.logIDs...)
+			wantLogs := make(map[int64]bool)
+			for _, logID := range tc.wantLogIDs {
+				wantLogs[logID] = true
+			}
+
+			for logID, enabled := range tc.set {
+				if err := m.SetWritable(ctx, directoryID, logID, enabled); err != nil {
+					t.Errorf("SetWritable(): %v", err)
+				}
+			}
+
+			logs := make(map[int64]bool)
+			// Run enough times to ensure that random sampling will get us all logs.
```

Comment:
could you instead set all logs not-writable except for one and ensure that is the one that gets written to?

Reply:
Sure